includes/class-content-distribution.php

```diff
@@ -0,0 +1,422 @@
+		if ( 'network_post_updated' === $action_name ) {
+			return 1;
+		}
+		return $priority;
+	}
+
+	/**
+	 * Post update listener callback.
+	 *
+	 * @param WP_Post|int $post The post object or ID.
+	 *
+	 * @return array|null The post payload or null if the post is not distributed.
+	 */
+	public static function handle_post_updated( $post ) {
+		if ( ! self::is_post_distributed( $post ) ) {
+			return;
+		}
+
+		$payload = self::get_post_payload( $post );
+		if ( is_wp_error( $payload ) ) {
+			return;
+		}
+
+		return $payload;
+	}
+
+	/**
+	 * Linked post inserted listener callback.
+	 *
+	 * @param int     $post_id      The linked post ID.
+	 * @param boolean $is_unlinked  Whether the post is unlinked.
+	 * @param array   $post_payload The post payload.
+	 */
+	public static function handle_linked_post_inserted( $post_id, $is_unlinked, $post_payload ) {
+		return [
+			'origin'      => [
+				'site_url' => $post_payload['site_url'],
+				'post_id'  => $post_payload['post_id'],
+			],
+			'destination' => [
+				'site_url'    => get_bloginfo( 'url' ),
+				'post_id'     => $post_id,
+				'is_unlinked' => $is_unlinked,
+			],
+		];
+	}
+
+	/**
+	 * Get the post types that are allowed to be distributed across the network.
+	 *
+	 * @return array Array of post types.
+	 */
+	public static function get_distributed_post_types() {
+		/**
+		 * Filters the post types that are allowed to be distributed across the network.
+		 *
+		 * @param array $post_types Array of post types.
+		 */
+		return apply_filters( 'newspack_network_distributed_post_types', [ 'post' ] );
+	}
+
+	/**
+	 * Set the distribution configuration for a given post.
+	 *
+	 * @param int   $post_id   The post ID.
+	 * @param int[] $site_urls Array of site URLs to distribute the post to.
+	 *
+	 * @return void|WP_Error Void on success, WP_Error on failure.
+	 */
+	public static function set_post_distribution( $post_id, $site_urls = [] ) {
+		$post = get_post( $post_id );
+		if ( ! $post ) {
+			return new WP_Error( 'invalid_post', __( 'Invalid post.', 'newspack-network' ) );
+		}
+		$config = get_post_meta( $post_id, self::DISTRIBUTED_POST_META, true );
+		if ( ! is_array( $config ) ) {
+			$config = [];
+		}
+		// Set post hash to link the post across the network.
+		if ( empty( $config['post_hash'] ) ) {
+			$config['post_hash'] = wp_generate_password( 32, false );
+		}
+		$config['enabled']   = empty( $site_urls ) ? false : true;
+		$config['site_urls'] = $site_urls;
+		update_post_meta( $post_id, self::DISTRIBUTED_POST_META, $config );
+	}
+
+	/**
+	 * Manually trigger post distribution.
+	 *
+	 * @param WP_Post|int $post The post object or ID.
+	 */
+	public static function distribute_post( $post ) {
+		if ( ! self::is_post_distributed( $post ) ) {
+			return new WP_Error( 'post_not_distributed', __( 'The post is not distributed across the network.', 'newspack-network' ) );
+		}
+		Data_Events::dispatch( 'network_post_updated', self::get_post_payload( $post ) );
```

Comment:
Maybe get the payload from `handle_post_update` so you have one single source of truth about the data payload. (If anything changes in that method, you might end up with discrepancies between the two ways this event is dispatched